spring-cql/src/test/java/org/springframework/cassandra/config/PoolingOptionsFactoryBeanUnitTests.java

```diff
@@ -0,0 +1,229 @@
+		poolingOptionsFactoryBean.setIdleTimeoutSeconds(120);
+		poolingOptionsFactoryBean.setInitializationExecutor(mockExecutor);
+		poolingOptionsFactoryBean.setLocalCoreConnections(50);
+		poolingOptionsFactoryBean.setLocalMaxConnections(1000);
+		poolingOptionsFactoryBean.setLocalMaxSimultaneousRequests(200);
+		poolingOptionsFactoryBean.setLocalMinSimultaneousRequests(100);
+		poolingOptionsFactoryBean.setPoolTimeoutMilliseconds(300);
+		poolingOptionsFactoryBean.setRemoteCoreConnections(25);
+		poolingOptionsFactoryBean.setRemoteMaxConnections(250);
+		poolingOptionsFactoryBean.setRemoteMaxSimultaneousRequests(100);
+		poolingOptionsFactoryBean.setRemoteMinSimultaneousRequests(50);
+
+		assertThat(poolingOptionsFactoryBean.getHeartbeatIntervalSeconds(), is(equalTo(15)));
+		assertThat(poolingOptionsFactoryBean.getIdleTimeoutSeconds(), is(equalTo(120)));
+		assertThat(poolingOptionsFactoryBean.getInitializationExecutor(), is(equalTo(mockExecutor)));
+		assertThat(poolingOptionsFactoryBean.getLocalCoreConnections(), is(equalTo(50)));
+		assertThat(poolingOptionsFactoryBean.getLocalMaxConnections(), is(equalTo(1000)));
+		assertThat(poolingOptionsFactoryBean.getLocalMaxSimultaneousRequests(), is(equalTo(200)));
+		assertThat(poolingOptionsFactoryBean.getLocalMinSimultaneousRequests(), is(equalTo(100)));
+		assertThat(poolingOptionsFactoryBean.getPoolTimeoutMilliseconds(), is(equalTo(300)));
+		assertThat(poolingOptionsFactoryBean.getRemoteCoreConnections(), is(equalTo(25)));
+		assertThat(poolingOptionsFactoryBean.getRemoteMaxConnections(), is(equalTo(250)));
+		assertThat(poolingOptionsFactoryBean.getRemoteMaxSimultaneousRequests(), is(equalTo(100)));
+		assertThat(poolingOptionsFactoryBean.getRemoteMinSimultaneousRequests(), is(equalTo(50)));
+	}
+
+	@Test
+	public void afterPropertiesSetInitializesLocalPoolingOptions() throws Exception {
+		PoolingOptionsFactoryBean poolingOptionsFactoryBean = new PoolingOptionsFactoryBean() {
+			@Override PoolingOptions newPoolingOptions() {
+				return mockPoolingOptions;
+			}
+		};
+
+		when(mockPoolingOptions.getNewConnectionThreshold(eq(HostDistance.LOCAL))).thenReturn(1);
+
+		poolingOptionsFactoryBean.setHeartbeatIntervalSeconds(60);
+		poolingOptionsFactoryBean.setIdleTimeoutSeconds(300);
+		poolingOptionsFactoryBean.setLocalCoreConnections(10);
+		poolingOptionsFactoryBean.setLocalMaxConnections(100);
+		poolingOptionsFactoryBean.setLocalMaxSimultaneousRequests(50);
+		poolingOptionsFactoryBean.setLocalMinSimultaneousRequests(5);
+		poolingOptionsFactoryBean.setPoolTimeoutMilliseconds(180);
+
+		assertThat(poolingOptionsFactoryBean.getObject(), is(nullValue(PoolingOptions.class)));
+
+		poolingOptionsFactoryBean.afterPropertiesSet();
+
+		assertThat(poolingOptionsFactoryBean.getObject(), is(sameInstance(mockPoolingOptions)));
+		assertThat(poolingOptionsFactoryBean.getObjectType(), is(equalTo((Class) mockPoolingOptions.getClass())));
+
+		InOrder inOrder = inOrder(mockPoolingOptions);
```

Comment:
`InOrder` tends to cause a lot of maintenance once tests are in place. We should have some tests which take real XML config and create the `PoolingOptions`. Who knows what happens inside of `PoolingOptions` in the future.